core/domain/learner_progress_services.py

```diff
@@ -616,3 +648,118 @@ def get_collection_summary_dicts(collection_summaries):
         })
 
     return summary_dicts
+
+
+def get_activity_progress(user_id):
+    """Returns the progress of the learners - the explorations and collections
+    completed by the user and those in progress.
+
+    Args:
+        user_id: str. The id of the learner.
+
+    Returns:
```

Comment:
The return type here is a 6-tuple, right? That is the type that should be used:

```
    Returns:
        tuple. A 6-tuple where each element represents the following:
            ...
```

In fact, it would probably be better to make the return value a LearnerProgress domain object instead. It's complicated enough to be its own "thing", IMO.